bin/docker-test-proxy

```diff
@@ -10,8 +10,10 @@ fi
 bindir="$( cd "$( dirname "${BASH_SOURCE[0]}" )" && pwd )"
 rootdir="$( cd $bindir/.. && pwd )"
 
+
 docker build -f $rootdir/proxy/Dockerfile . \
     --target build \
     -t proxy-build \
-    --build-arg=PROXY_UNOPTIMIZED=1 &&
+    --build-arg=PROXY_UNOPTIMIZED=1 > /dev/null
+echo "running proxy tests in docker image..."
```

Comment:
strong preference that we not echo things like this. if we do, should go to stderr and not stdout.

Reply:
Okay, I thought that if we silenced the docker build output, it was good to have some indication that the tests were run in a container rather than on the host, since `docker run -it`'s pseudo-TTY colors the test output and it looks the same as a `cargo test` on the host.

Reply:
I think f002f73134b298413e92fcdf87e78a3d600bd5ce, which changes this to just honor the `DOCKER_TRACE` env var, is a reasonable compromise, WDYT?

Reply:
(sorry for making substantial changes post-review-approval)

Reply:
If i'm running bin/docker-test-proxy, I expect it to run in docker ;)

It's always my preference that tools be as terse as possible and only output information essential to the job. (i.e. if i run this tool 30x a day, and it just prints out this static message every time, i'm going to just ignore it). Furthermore, I generally want stdout to have only the actionable/parseable things.

If anything, we would maybe want to echo _before_ running `docker build` (which may take quite a while). This sort of thing sounds perfect for the `log_debug` function https://github.com/runconduit/conduit/blob/master/bin/_log.sh#L11 which hides output by default but can be enabled with `BUILD_DEBUG=1`.